components/brave_wallet/browser/eth_tx_controller_unittest.cc

```diff
@@ -5,13 +5,310 @@
+  EXPECT_TRUE(tx_meta);
+
+  uint256_t gas_price_value;
+  uint256_t gas_limit_value;
+  // Gas price should be filled by requesting eth_gasPrice.
+  EXPECT_TRUE(HexValueToUint256("0x595698248593c0000", &gas_price_value));
+  EXPECT_TRUE(HexValueToUint256(gas_limit, &gas_limit_value));
+  EXPECT_EQ(tx_meta->tx->gas_price(), gas_price_value);
+  EXPECT_EQ(tx_meta->tx->gas_limit(), gas_limit_value);
+
+  SetErrorInterceptor();
+  callback_called = false;
+  eth_tx_controller_->AddUnapprovedTransaction(
+      std::move(tx_data), from(),
+      base::BindOnce(&AddUnapprovedTransactionFailureCallback,
+                     &callback_called));
+  base::RunLoop().RunUntilIdle();
+  EXPECT_TRUE(callback_called);
+}
+
+TEST_F(EthTxControllerUnitTest,
+       AddUnapprovedTransactionWithoutGasPriceAndGasLimit) {
+  std::vector<uint8_t> data;
+  EXPECT_TRUE(base::HexStringToBytes("010200", &data));
+  auto tx_data = mojom::TxData::New(
+      "0x06", "" /* gas_price */, "" /* gas_limit */,
+      "0xbe862ad9abfe6f22bcb087716c7d89a26051f74c", "0x016345785d8a0000", data);
+  bool callback_called = false;
+  std::string tx_meta_id;
+
+  eth_tx_controller_->AddUnapprovedTransaction(
+      tx_data.Clone(), from(),
+      base::BindOnce(&AddUnapprovedTransactionSuccessCallback, &callback_called,
+                     &tx_meta_id));
+
+  base::RunLoop().RunUntilIdle();
+  EXPECT_TRUE(callback_called);
+  auto tx_meta = eth_tx_controller_->GetTxForTesting(tx_meta_id);
+  EXPECT_TRUE(tx_meta);
+
+  uint256_t gas_price_value;
+  uint256_t gas_limit_value;
+  EXPECT_TRUE(HexValueToUint256("0x595698248593c0000", &gas_price_value));
+  EXPECT_TRUE(HexValueToUint256("0x9604", &gas_limit_value));
+  EXPECT_EQ(tx_meta->tx->gas_price(), gas_price_value);
+  EXPECT_EQ(tx_meta->tx->gas_limit(), gas_limit_value);
+
+  SetErrorInterceptor();
+  callback_called = false;
+  eth_tx_controller_->AddUnapprovedTransaction(
+      std::move(tx_data), from(),
+      base::BindOnce(&AddUnapprovedTransactionFailureCallback,
+                     &callback_called));
+  base::RunLoop().RunUntilIdle();
+  EXPECT_TRUE(callback_called);
+}
+
+TEST_F(EthTxControllerUnitTest,
+       AddUnapprovedTransactionWithoutGasPriceAndGasLimitForEthSend) {
+  auto tx_data =
+      mojom::TxData::New("0x06", "" /* gas_price*/, "" /* gas_limit */,
+                         "0xbe862ad9abfe6f22bcb087716c7d89a26051f74c",
+                         "0x016345785d8a0000", std::vector<uint8_t>());
+  bool callback_called = false;
+  std::string tx_meta_id;
+
+  eth_tx_controller_->AddUnapprovedTransaction(
+      std::move(tx_data), from(),
+      base::BindOnce(&AddUnapprovedTransactionSuccessCallback, &callback_called,
+                     &tx_meta_id));
+
+  base::RunLoop().RunUntilIdle();
+  EXPECT_TRUE(callback_called);
+  auto tx_meta = eth_tx_controller_->GetTxForTesting(tx_meta_id);
+  EXPECT_TRUE(tx_meta);
+
+  // Default value will be used.
+  EXPECT_EQ(tx_meta->tx->gas_price(), 150u * 1e9);
```

Comment:
maybe ull here and below?

Reply:
Use the defined constant now in https://github.com/brave/brave-core/pull/10131/commits/642734ae3d2a7c4e088c8274d06d34db7bbd2986.